File: solutions/ruby/prime-factors/1/prime_factors.rb
# PrimeFactors
module PrimeFactors
  module_function

  def for(n)
    [].tap do |factors|
      factor = 2
      while n > 1
        while (n % factor).zero?
          factors << factor
          n /= factor
        end
        factor += 1
      end
    end
  end
end

File: solutions/clojure/prime-factors/1/src/prime_factors.clj
(ns prime-factors)

(defn of [n]
  (loop [n n
         factor 2
         factors []]
    (cond
      (= n 1) factors
      (zero? (mod n factor)) (recur (quot n factor) factor (conj factors factor))
      :else (recur n (inc factor) factors))))
